<?xml version="1.0"?>
<launch>
  <arg name="ojama" default="true"/>
  <arg name="beego" default="true"/>
  <arg name="rviz" default="true"/>

  <!-- Yamabiko -->
    <node pkg="ypspur_ros" type="ypspur_ros" name="ypspur_ros">
      <param name="param_file" value="/home/ruan-x/researches/programs/platform/yp-robot-params/robot-params/speego.param" />
      <param name="port" value="/dev/ttyACM1" />
      <param name="compatible" value="1"/>
    </node>
    <!-- urg -->
    <node name="hokuyo" pkg="hokuyo_node" type="hokuyo_node" respawn="false" output="screen">
      <param name="serial_port" value="/dev/ttyACM0" />
      <param name="angle_min" value="-1.5707963" />
      <param name="angle_max" value="1.5707963" />
    </node>

    <!--ここからgmappingのコード記入-->
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" clear_params="true">
      <rosparam command="load" file="$(find yamasemi_sim)/config/gmapping.yaml" />
      <!-- <remap from="scan" to="scan" /> -->
      <param name="base_frame" value="base_link" />
      <param name="odom_frame" value="odom" />
      <param name="map_frame"  value="map" />
      <!-- 以下gmappingのパラメータ調整 -->
    </node> 

     <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser"
    args=" 0.2245 0 0.2 0.0 0 0 /base_link /laser 10" />
    <!--ここまで-->
    <!-- joy_nodeの起動 -->
    <node pkg="joy" type="joy_node" name="joy_node" output="screen">
    </node>

    <node name="joy_controller" pkg="final_task_rxy" type="joy_controller">
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find yamasemi_tf2)/rviz/yamabico.rviz" />
  

</launch>
